servers/visual/visual_server_scene.cpp

```diff
@@ -1398,9 +1398,9 @@ bool VisualServerScene::_light_instance_update_shadow(Instance *p_instance, cons
 
 				if (p_cam_orthogonal) {
 
-					float w, h;
-					p_cam_projection.get_viewport_size(w, h);
-					camera_matrix.set_orthogonal(w, aspect, distances[(i == 0 || !overlap) ? i : i - 1], distances[i + 1], false);
+					Vector2 vp_he = p_cam_projection.get_viewport_half_extents();
+
+					camera_matrix.set_orthogonal(vp_he.y * 2.0, aspect, distances[(i == 0 || !overlap) ? i : i - 1], distances[i + 1], false);
```

Comment:
Any reason why it went from using width in the original to using height in the updated code? I feel like they may not result in the same behaviour.

Reply:
Yes .. they don't result in the same behaviour! The old code was broken badly. #26637 

This was the original reason for the PR, the use of height (and the 2x multiplier) fixes the bug.

It turns out that when you pass size into CameraMatrix::set_orthogonal, size only persists in terms of the height dimension (unless the aspect ratio is 1.0). To get back to the original size in order to create a new CameraMatrix, you have to use the HEIGHT rather than the width. I.e. height is preserved rather than width. This makes sense for typical use cases of aspect ratio.